### ISSUE TITLE
div with tabindex="-1" can't receive keyboard events in IEDriverServer 3.12.0

### ISSUE THREAD

**Issue by mtrea** [SeleniumHQ/selenium#6038]
## Meta -
OS:  Windows 7
Selenium Version:  3.11, IEDriverServer 3.12.0
Browser:  Internet Explorer
Browser Version:  11.0.9600.17126

## Expected Behavior -
In IEDriverServer 3.11.1, I can have something in the DOM like:
`<div id="target" tabindex="-1">click</div>`
and call:
`self.driver.find_element_by_id("target").send_keys(Keys.ENTER)`

## Actual Behavior -
When I updated to IEDriverServer 3.12.0, the above behavior was regressed. Now, when I try to call send_keys, I get an ElementNotInteractableException with the message:
> Element cannot be interacted with via the keyboard because it is not focusable

Note that changing tabindex to "0" does not yield an ElementNotInteractableException.

## Steps to reproduce -
Page source: https://jsfiddle.net/6r7dtk3h/

Assuming your test class has setup code that instantiates a driver for IE and gets the above page, run the following test case:
```
from selenium.webdriver.common.keys import Keys

  def testClick(self):
    self.driver.find_element_by_id("target").click()
    self.driver.find_element_by_id("target").send_keys(Keys.ENTER)
```

This should demonstrate the behavior documented above. The click will succeed, but send_keys will yield an ElementNotInteractableException.

**Comment by muli09** [comment]
I have a similar issue but my was selenium is performed but in application it not performed action 
<div role="dialog" tabindex="-1" >
<button >delete</button>

<.div>

**Comment by linghuaj** [comment]
any updates on this bug?

**Comment by ketoor** [comment]
I guess, the problem was introduced in "Adding IsFocusable method to IE driver Element class" (eda10eed6a193f47224c1e1237c636d924ece435) by @jimevans 
We have found same problem and we are forced to downgrade Selenium to 3.11.
Do you know, where is the source code for generated files (like atoms.h)?

**Comment by jimevans** [comment]
@ketoor The automation atoms for the IE driver are compiled from the `javascript` directory of the project repository. In the case of the `isFocusable` function, it’s `bot.dom.isFocusable`, and can be found in [dom.js](https://github.com/SeleniumHQ/selenium/blob/master/javascript/atoms/dom.js).

**Comment by gjyalpha** [comment]
The issue is that in [W3C WebDriver Specification](https://w3c.github.io/webdriver/), it refers to whatwg [focusable](https://html.spec.whatwg.org/#focusable).

```
All elements that are focusable areas are said to be focusable.

There are two special types of focusability for focusable areas:

- A focusable area is said to be sequentially focusable if it is included in its Document's sequential focus navigation order and the user agent determines that it is sequentially focusable.
- A focusable area is said to be click focusable if the user agent determines that it is click focusable. User agents should consider focusable areas with non-null tabindex values to be click focusable.
```

`bot.dom.isFocusable()` didn't declare which focusable, sequentially focusable or click focusable, it's supposed to be. But it seems the implementation is sequentially focusable. It sounds to me that it should return `true` for both.

And W3c WebDriver Spec says:

```
The Element Send Keys command scrolls into view the form control element and then sends the provided keys to the element. In case the element is not keyboard-interactable, an element not interactable error is returned.

A keyboard-interactable element is any element that has a focusable area ...
```

So it sounds to me it allows send keys to tabindex="-1" element, i.e. click focusable element.

**Comment by mtrea** [comment]
Ack.

**Comment by diemol** [comment]
What happens if you try the IE mode in Edge?

**Comment by mtrea** [comment]
Sorry, I can't answer that. Our infrastructure doesn't support Edge at the moment.

**Comment by diemol** [comment]
IE won't be supported anymore soon, which is why I asked.

https://docs.microsoft.com/en-us/microsoft-edge/webdriver-chromium/ie-mode?tabs=c-sharp

**Comment by titusfortner** [comment]
The latest version of IE driver fixed some bugs, please try with IE Driver 4.0
https://github.com/SeleniumHQ/selenium/releases/tag/selenium-4.0.0

**Comment by diemol** [comment]
I will close this as IE should be used only under IE mode in Edge these days. If there are any issues, they should be raised to the Edge folks after testing in IE mode.
